airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -266,6 +279,69 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         params = {"include": "description"}
         yield from self.read(partial(self._api_get, url="tickets"), params=params)
 
+    @staticmethod
+    def get_tickets(
+        result_return_limit: int, getter: Callable, params: Mapping[str, Any] = None, ticket_paginate_limit: int = 300
+    ) -> Iterator:
+        """
+        Read using getter
+
+        This block extends TicketsAPI Stream to overcome '300 page' server error.
+        Since the TicketsAPI Stream list has a 300 page pagination limit, after 300 pages, update the parameters with
+        query using 'updated_since' = last_record, if there is more data remaining.
+        """
+        params = params or {}
+        # the maximum page allowed to pull during pagination.
+        # ticket_paginate_limit 300
+        # Start page
+        page = 1
+        # Initial request parameters
+        params = {**params, "per_page": result_return_limit}
+
+        while True:
+            params["page"] = page
+            batch = list(getter(params=params))
+            yield from batch
+
+            if len(batch) < result_return_limit:
+                return iter(())
+
+            # get last_record from latest batch, pos. 0, because of DESC order of records
+            last_record = batch[0]["updated_at"]
+
+            # checkpoint & switch the pagination
+            if page == ticket_paginate_limit:
+                page = 0  # reset page counter
+                # Adding +1 second to last_record value, to avoid record duplication,
+                # this potentialy could lead to skipping some records during the fetch,
+                # in cases where multiple records are created at the same datetime, at the level of seconds
+                # this behaviour is tested in test_300_page.py unit_test (test1)
+                # to avoid such ocations, please comment or remove the `.add(seconds=1)' part.
+                last_record = pendulum.parse(last_record).add(seconds=1)
+                # updating request parameters with last_record state
+                params["order_by"] = "updated_at"
+                params["updated_since"] = last_record
```

Comment:
https://developers.freshdesk.com/api/#list_all_tickets
Actually, the fix does not solve the current problem because API returns the most recent tickets first by default:

Default sort order is created_at
Default sort order type is desc

So if we call "updated_since=2015-01-19T02:00:00Z"  - it will return newer tickets, not the oldest as we currently expect.


So in order to fix the problem and use 'updated_since' filter, we need use:
1. sort order = 'updated_at'  (instead of default 'created_at')
2. order_type=asc             (instead of default 'desc')
for all requests

so, in this situation, API will return the oldest ticket first and if the limit is reached then we can additionally call ' "updated_since=' to get the newest tickets.

Reply:
Test should be updated - change the order in tickets examples (the newest should go first)

Reply:
First things first:
The spec of this connector doesn't require the start_date. I don't know why, but it just a fact. 

No problem, to change the scope of data retrieval for tickets stream, what about others?
This whole implementation looks like needs to be changed?

If you look at the incremental mode, you will see the difference, where incrementally we fetch more data than using full-refresh, particularly for the tickets stream this is case.

Reply:
Checked the whole implementation uses the `updated_at` so decided to leave it by default for all streams. 
Also, added the `order_type` = "asc" parameter to TicketsApi stream in order to fetch the oldest records first.
Tests are passing as well.